### PR TITLE
Update implementation for Android 11, drop old support.

### PR DESCRIPTION
**Pull request by jusa**

### CODE REVIEW

File: rpm/pulseaudio-modules-droid.spec [hunk 1]
```diff
@@ -73,8 +73,10 @@ rm -rf %{buildroot}
 %files devel
 %defattr(-,root,root,-)
 %dir %{_prefix}/include/pulsecore/modules/droid
```
Comment (Thaodan) [inline rpm/pulseaudio-modules-droid.spec:75]: Can you use %_includedir instead?

File: rpm/pulseaudio-modules-droid.spec [hunk 2]
```diff
@@ -51,7 +51,7 @@ else
 . %{_libdir}/droid-devel/hw-release.vars
 fi
 %reconfigure --disable-static --with-droid-device=$MER_HA_DEVICE
-make %{?jobs:-j%jobs}
+%make_build
 
 %install
 rm -rf %{buildroot}
```
Comment (Thaodan) [inline rpm/pulseaudio-modules-droid.spec:57]: This is redundant.

**Comment by Thaodan** [comment]
Please also adjust the readme to only list Android 11.

File: src/common/config-parser-xml.c [hunk 3]
```diff
@@ -953,133 +955,184 @@ static bool parse_file(struct parser_data *data, const struct element_parser *ro
...
-                        hw_module->global_config->default_output_device |= device_port->type;
-                    break;
-                } else
-                    pa_log_info("Couldn't find matching <" ELEMENT_devicePort " tagName=%s> for <" ELEMENT_mixPort " name=%s>",
-                                route->sink, source->name);
-            }
+    c_module = pa_xnew0(dm_config_module, 1);
+    c_module->config = config;
+    c_module->name = pa_xstrdup(module->name);
+    c_module->version_major = 0; /* AUDIO_API_VERSION_GET_MAJ(device->common.version) */
```
Comment (jusa) [inline src/common/config-parser-xml.c:1023]: Remove comments, just use the values.

File: src/common/droid-util.c [hunk 4]
```diff
@@ -70,35 +71,44 @@
...
+    { "input_atoi",             DM_OPTION_INPUT_ATOI            },
+    { "close_input",            DM_OPTION_CLOSE_INPUT           },
+    { "unload_no_close",        DM_OPTION_UNLOAD_NO_CLOSE       },
+    { "hw_volume",              DM_OPTION_HW_VOLUME             },
+    { "output_make_writable",   DM_OPTION_OUTPUT_MAKE_WRITABLE  },
+    { "realcall",               DM_OPTION_REALCALL              },
+    { "unload_call_exit",       DM_OPTION_UNLOAD_CALL_EXIT      },
+    { "output_fast",            DM_OPTION_OUTPUT_FAST           },
+    { "output_deep_buffer",     DM_OPTION_OUTPUT_DEEP_BUFFER    },
+    { "audio_cal_wait",         DM_OPTION_AUDIO_CAL_WAIT        },
```
Comment (jusa) [inline src/common/droid-util.c]: audio_cal_wait, standby_set_route, speaker_before_voice are not needed with Android 11 adaptations (at least for the moment).

File: src/common/droid-util.c [hunk 5]
```diff
@@ -1268,247 +1193,369 @@ static int stream_standby(pa_droid_stream *s) {
...
+        goto fail;
+
     pa_droid_hw_module_lock(module);
     ret = module->device->open_output_stream(module->device,
-                                             ++module->stream_out_id,
-                                             devices,
-                                             module_output->flags,
+                                             ++module->stream_id,
+                                             device_port->type,
+                                             mix_port->flags,
```
Comment (jusa) [inline src/common/droid-util.c]: Should use flags.

File: src/common/utils.c [hunk 6]
```diff
@@ -0,0 +1,43 @@
+#ifdef HAVE_CONFIG_H
```
Comment (mlehtima) [inline src/common/utils.c:1]: Missing copyright header.

File: src/common/include/droid/utils.h [hunk 7]
```diff
@@ -0,0 +1,11 @@
+#ifndef foodroidcommonutilsfoo
```
Comment (mlehtima) [inline src/common/include/droid/utils.h:1]: Missing copyright header.

**Comment by mlehtima** [comment]
Copyright year changes seem to be to 2021, it's already 2022. Also some changed files didn't have change to copyright years.

**Review by mlehtima — APPROVED** [review]

**Review by Thaodan — APPROVED** [review]
LGTM

**Pull request merged**